### PR TITLE
Deploy the app

### PR DESCRIPTION
**Pull request by sofita**
## Context
We need to show the app on a domain

## Changes
- Pulled everything in `server/` into the root directory
- Renamed frontend code folder from `cha-ching-app` to `client`
- Bundled the .env variables into one .env
- 

## Pre-merge checklist
- [x] run backend tests
- [x] run frontend tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,3 +5,9 @@ A virtual payment method between a web application and a customer/user to allow
 # Technologies
 
 Javascript, React, Stripe, Jest
+
+## Deploying the Server
+
```
Comment (sofita) [inline README.md:10]:
```suggestion
We deploy the app with [heroku](https://www.heroku.com). the front and back end are deployed separately. 
```

File: app.js [hunk 2]
```diff
@@ -147,6 +146,12 @@ app.get("/stripe-client-secret", enforceSignature, async (req, res) => {
   res.json(paymentIntentResult);
 });
 
+// All remaining requests return the React app, so it can handle routing.
+// app.get("*", function (request, response) {
+//   response.sendFile(path.resolve(__dirname, "../client/build", "index.html"));
+// });
+
```
Comment (sofita) [inline app.js:153]:
```suggestion
```

File: client/README.md [hunk 3]
```diff
@@ -68,3 +68,9 @@ This section has moved here: [https://facebook.github.io/create-react-app/docs/d
 ### `npm run build` fails to minify
 
 This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+
+## Deploying the Client
+
```
Comment (sofita) [inline client/README.md:73]:
```suggestion
We deploy the app with [heroku](https://www.heroku.com). The front and back end are deployed separately. 
```

File: client/src/__snapshots__/ElementsProviderCheckout.test.js.snap [hunk 4]
```diff
@@ -67,3 +67,41 @@ exports[`<ElementsProviderCheckout/> renders Stripe's Elements provider with cli
   </div>
 </DocumentFragment>
 `;
+
+exports[`<ElementsProviderCheckout/> renders payment form when clientSecret is present 1`] = `
```
Comment (sofita) [inline client/src/__snapshots__/ElementsProviderCheckout.test.js.snap:71]: look into test that was probably duplicated, we shouldn't have 2 snapshots for this test case

File: package.json [hunk 5]
```diff
@@ -9,7 +9,9 @@
   "scripts": {
     "test": "jest",
     "test:watch": "jest --watch",
-    "start": "nodemon node app.js"
+    "debug": "nodemon node app.js",
```
Comment (sofita) [inline package.json:12]:
```suggestion
    "dev": "nodemon node app.js",
```

Comment (Ashlynn223) [reply package.json:12]: What do you think of changing the names from degug and start to be startproduction and startdev

File: package.json [hunk 6]
```diff
@@ -9,7 +9,9 @@
...
+    "start": "node app.js",
+    "build": "cd client/ && npm install && npm run build"
```
Comment (sofita) [inline package.json:14]: Maybe can remove it

Comment (sofita) [reply package.json]: removed

File: client/src/__snapshots__/ElementsProviderCheckout.test.js.snap (hunk 4 above)
Comment (sofita) [reply client/src/__snapshots__/ElementsProviderCheckout.test.js.snap:35]: done

**Review by Ashlynn223 — APPROVED** [review]
Looks great!

**Pull request merged**